src/foam/core/CSV.js

```diff
@@ -0,0 +1,359 @@
+/**
+ * @license
+ * Copyright 2017 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.csv',
+  name: 'Outputter',
+
+  documentation: 'CSV Outputter.',
+
+  properties: [
+    {
+      class: 'String',
+      name: 'buf_',
+      value: ''
+    },
+    {
+      class: 'Boolean',
+      name: 'includeQuotes',
+      help: 'If true, values are quoted with the `"` char',
+      value: true
+    },
+    {
+      class: 'String',
+      name: 'delimiter',
+      value: ','
+    },
+    {
+      class: 'String',
+      name: 'nestedObjectSeperator',
+      value: '__'
+    },
+    {
+      class: 'StringArray',
+      name: 'nestedObjectNames'
```

Comment:
Updating.

Is there a way to pass in a param into a `mmethod` (I'll need to pass in the `nestedObjectNames` stack so that it is preserved through recursive calls.